samples/usb/main.c

```diff
@@ -0,0 +1,38 @@
+#include <hal/debug.h>
+#include <hal/video.h>
+#include <windows.h>
+#include <usbh_lib.h>
+
+/*
+ * This demo will initialise the usb stack, detect events on the usb ports
+ * and print the device PID and VID.
+*/
+void device_connection_callback(UDEV_T *udev, int status)
+{
+    debugPrint("Device connected on port %u (PID: %04x VID: %04x)\n", udev->port_num,
+                                                                      udev->descriptor.idProduct,
+                                                                      udev->descriptor.idVendor);
+}
+
+void device_disconnect_callback(UDEV_T *udev, int status)
+{
+     debugPrint("Device disconnected on port %u (PID: %04x VID: %04x)\n", udev->port_num,
+                                                                          udev->descriptor.idProduct,
+                                                                          udev->descriptor.idVendor);
+}
+
+int main(void)
+{
+    XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+
+    usbh_core_init();
+    usbh_install_conn_callback(&device_connection_callback, &device_disconnect_callback);
+    debugPrint("== ochi usb test ==\n");
+    debugPrint("== Plug and unplug USB devices to test ==\n");
+    while (1) {
+        usbh_pooling_hubs();
+        Sleep(1);

```

Comment:
nit: unusual short sleeping period. Other samples either wait for vblank or wait i.e. 1000ms.

Reply:
Although the low level events happen via interrupt, the callbacks and overall library updates occur when usbh_pooling_hubs is called so this should be called reasonably often in this sample to illustrate quick updates when hot plugging.
The 1ms is just based on USB frame size.

I'd rather remove it all together which I can do if it causes some confusion. Thanks for your review!